.github/workflows/lint.yml

```diff
@@ -96,3 +96,17 @@ jobs:
 
       - name: Run license checks
         run: addlicense -check -ignore '.git/**' -ignore 'build/**' -ignore 'llvm*/**' .
+
```

Comment:
Please add commit description to the PR (adding as comment to block)

Reply:
Done, sorry about that. I originally felt that the PR kind of explained itself, but seeing the comments from you and Farid I realized that was not the case, there's actually quite a lot going on here. PTAL at the PR description and let me know if you have any questions 🙏